pkg/secrets/k8s_secrets_storage/provide_conjur_secrets_test.go

```diff
@@ -383,3 +387,91 @@ func TestProvide(t *testing.T) {
 		}
 	}
 }
+
+func Test_secrets_contentChanges(t *testing.T) {
+
+	var desc               string
+	var k8sSecrets         k8sStorageMocks.K8sSecrets
+	var requiredSecrets    []string
+	var denyConjurRetrieve bool
+	var denyK8sRetrieve    bool
+	var denyK8sUpdate      bool
+
+	desc = "Only update secrets when there are changes"
+	// Initial case, k8s secret should be updated
+	k8sSecrets = k8sStorageMocks.K8sSecrets{
+		"k8s-secret1": {
+			"conjur-map": {"secret1": "conjur/var/path1"},
+		},
+	}
+	requiredSecrets = []string{"k8s-secret1"}
+	mocks := newTestMocks()
+	mocks.setPermissions(denyConjurRetrieve, denyK8sRetrieve, denyK8sUpdate)
+	for secretName, secretData := range k8sSecrets {
+		mocks.kubeClient.AddSecret(secretName, secretData)
+	}
+	provider := mocks.newProvider(requiredSecrets)
+	err := provider.Provide()
+	assert.False(t, mocks.logger.InfoWasLogged(messages.CSPFK020I))
+	assertSecretsUpdated(
+		expectedK8sSecrets{
+			"k8s-secret1": {"secret1": "secret-value1"},
+		},
+		expectedMissingValues{} ) (t, mocks, err, desc)
+
+	// Call Provide again, verify is doesn't try to update the secret
+	// as there should be an error if it tried to write the secrets
+	denyK8sUpdate = true
+	mocks.setPermissions(denyConjurRetrieve, denyK8sRetrieve , denyK8sUpdate)
+	err = provider.Provide()
+	assert.NoError(t, err)
+	assert.True(t, mocks.logger.InfoWasLogged(messages.CSPFK020I))
+	// verify the same secret still exists
+	assertSecretsUpdated(
+		expectedK8sSecrets{
+			"k8s-secret1": {"secret1": "secret-value1"},
+		},
+		expectedMissingValues{} ) (t, mocks, err, desc)
```

Comment:
Something that might be worth considering here... this test involves a whole bunch of operations, and if when there's a test failure, it won't be immediately obvious which assert in this test is failing (and where in the sequence of operations it's failing), without having to match up line numbers in the source code. What you could do is to modify the `desc` string as you work your way through the sequence. That way, the failure summary for the assert that fails will provide us with details on where exactly it's failing.